### PR TITLE
Move network policy management before the first rolling udate

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Bugfix
- Enhancement / new feature

### Description

The manual rolling update is using the same algorithm as the main rolling update to find the leaders and use them to do the update in particular order. But for that it might need to have the right network policies to be able to access it.

This Pr moves the management of network policies before the manual rolling update. That should make sure that if the policies were not correct, they will be fixed and the rolling update can succeed.

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java [hunk 1]
```diff
@@ -183,16 +184,17 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
...
                 .compose(state -> state.zkNodesSecret())
-                .compose(state -> state.zkNetPolicy())
                 .compose(state -> state.zkPodDisruptionBudget())
                 .compose(state -> state.zkStatefulSet())
                 .compose(state -> state.zkScaleUp())
                 .compose(state -> state.zkRollingUpdate(this::dateSupplier))
                 .compose(state -> state.zkServiceEndpointReadiness())
                 .compose(state -> state.zkHeadlessServiceEndpointReadiness())
                 .compose(state -> state.zkPersistentClaimDeletion())
+
```
Comment (ppatierno) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: Remove this space?

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: Actually, I did there the blank line intentionally. It follows on the other blank lines which are already there and separate the different components - between the Kafka steps, TO steps and EO steps. So I just followed it and added steps between Kafka and Zoo.

Comment (ppatierno) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: yeah fine with it

Comment (scholzj) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java]: Thanks

**Pull request merged**